pkg/controller/installation/typha_autoscaler.go

```diff
@@ -39,16 +39,14 @@ const (
 
 // typhaAutoscaler periodically lists the nodes and, if needed, scales the Typha deployment up/down.
 // The number of Typha replicas depends on the number of nodes:
-// Nodes       Replicas
-//     1              1
-//     2              2
-//     3              3
-//   250              4
-//   500              5
-//  1000              6
-//  1500              7
-//  2000              8
-//  2000+            10
+//   Nodes          Replicas
+// <=    4                 1
```

Comment:
Just because we have a small cluster doesn't mean we shouldn't treat it as production. We should have HA in production, so as soon as we can have more than 1 typha we should. Ideally running 3 (when we can) so we have redundancy even when updating.

Reply:
I think we should do something like https://github.com/tigera/operator/pull/638 instead of having un-explained scalings of typha for the nodes. I think we should have some calculation based on the number of nodes. @caseydavenport WDYT?

Reply:
I think this is just a matter of being practical. In the case of <= 4 nodes, we can't afford to have 4 typhas running - that doesn't make any sense to me. Now, we could state that we should always have a minimum of 2 and ideally at least 3. But, what if we have <= 2 nodes? It seems like we should always try to have a *functional* cluster as our top priority followed closely by "production ready".

I don't really understand your comment about the "un-explained scalings". The new algorithm attempts to blend the practical with the ideal and I think it does a pretty decent job of balancing the two. That said, we might decide that we should adjust the #s slightly on the low end. For example:  <=2 (1 typha), <= 5 (2 typha) - however, we'll likely then need to add a toleration to the typha deployment so that it can run on master nodes.

Reply:
> // <=    4                 1

Isn't this suggesting that for any cluster less than 4 nodes we'll just use a single Typha? I think @tmjd meant that we should do something like this:

- 1 node -> 1 typha
- 2 nodes -> 2 typhas
- 3 nodes -> 3 typhas
- 4 nodes - > 3 typhas
- 5 nodes -> 3 typhas
. . .
- 250 nodes -> 4 typhas

Reply:
Yes I was suggesting we need 1:1, 2:2, 3:3, 4:3...
By un-explained I meant our other scaling levels at 250 and 500 and so on.